Tests/SourceKitTests/DocumentColorTests.swift

```diff
@@ -0,0 +1,152 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SKSupport
+import SKTestSupport
+import XCTest
+
+@testable import LanguageServerProtocol
+@testable import SourceKit
+
+final class DocumentColorTests: XCTestCase {
+  /// Connection and lifetime management for the service.
+  var connection: TestSourceKitServer! = nil
+
+  /// The primary interface to make requests to the SourceKitServer.
+  var sk: TestClient! = nil
+
+  /// The server's workspace data. Accessing this is unsafe if the server does so concurrently.
+  var workspace: Workspace! = nil
+
+  override func tearDown() {
+    workspace = nil
+    sk = nil
+    connection = nil
+  }
+
+  func initialize() {
+    connection = TestSourceKitServer()
+    sk = connection.client
+    let documentCapabilities = TextDocumentClientCapabilities()
+    _ = try! sk.sendSync(InitializeRequest(
+      processId: nil,
+      rootPath: nil,
+      rootURL: nil,
+      initializationOptions: nil,
+      capabilities: ClientCapabilities(workspace: nil, textDocument: documentCapabilities),
+      trace: .off,
+      workspaceFolders: nil))
+
+    workspace = connection.server!.workspace!
+  }
+
+  func getDocumentColorRequest(text: String) -> DocumentColorRequest {
```

Comment:
It's weird for this API to return the color request, but also implicitly open a document, which modifies the server state.  Please factor this so that the method either performs all the requests, or none of them.

Reply:
I also modified FoldingRangeTests and DocumentSymbolTests. Does it look good?

Reply:
Yes, thanks!